catroid/src/org/catrobat/catroid/ui/fragment/LookFragment.java

```diff
@@ -614,10 +615,22 @@ public void addLookDrawNewImage() {
 			return;
 		}
 
+		String[] lookNamesPresent = new String[100];
```

Comment:
Nowadays you hardly use arrays. You should prefer lists. Take a look at http://stackoverflow.com/a/1589952 for some reasons.

Creating an array with fixed size of `100` is not a good choice.

- Objects can have more than 100 looks
- and you also know how many looks the sprite has. `lookDataList.size()`

-> `new String[lookDataList.size()];`. However, you should always prefer (and like in this case use) some high-level container (like lists).